File: SYP.Web/Modules/Warehouse/Warehouses/WarehousesForm.cs
namespace SYP.Warehouse.Forms;

[FormScript("Warehouse.Warehouses")]
[BasedOnRow(typeof(WarehousesRow), CheckNames = true)]
public class WarehousesForm
{
    [HalfWidth]
    public string Code { get; set; }

    [HalfWidth]
    public string Name { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Address { get; set; }

    [HalfWidth]
    public string City { get; set; }

    [HalfWidth]
    public string Phone { get; set; }

    [HalfWidth, EmailAddressEditor]
    public string Email { get; set; }

    [HalfWidth]
    public string ManagerName { get; set; }

    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

    [HalfWidth]
    public bool IsActive { get; set; }

    [HalfWidth]
    public bool IsDefault { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Warehouse/WarehousesForm.ts
import { BooleanEditor, EmailAddressEditor, initFormType, PrefixedContext, StringEditor, TextAreaEditor } from "@serenity-is/corelib";

export interface WarehousesForm {
    Code: StringEditor;
    Name: StringEditor;
    Address: TextAreaEditor;
    City: StringEditor;
    Phone: StringEditor;
    Email: EmailAddressEditor;
    ManagerName: StringEditor;
    Description: TextAreaEditor;
    IsActive: BooleanEditor;
    IsDefault: BooleanEditor;
}

export class WarehousesForm extends PrefixedContext {
    static readonly formKey = 'Warehouse.Warehouses';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!WarehousesForm.init) {
            WarehousesForm.init = true;

            var w0 = StringEditor;
            var w1 = TextAreaEditor;
            var w2 = EmailAddressEditor;
            var w3 = BooleanEditor;

            initFormType(WarehousesForm, [
                'Code', w0,
                'Name', w0,
                'Address', w1,
                'City', w0,
                'Phone', w0,
                'Email', w2,
                'ManagerName', w0,
                'Description', w1,
                'IsActive', w3,
                'IsDefault', w3
            ]);
        }
    }
}